File: Advanced/PageAndRDL/CustomPdfExport/VB.NET/PdfRendering/Rendering/PdfContentGenerator.Image.vb
Imports PdfSharp.Drawing
Imports GrapeCity.ActiveReports.Drawing
Imports System.Drawing


Partial Class PdfContentGenerator
	Private Class Image
		Inherits ImageEx
		Private ReadOnly _imageInfo As XImage

		Public Sub New(imageInfo As XImage)
			_imageInfo = imageInfo
		End Sub

		Public Overrides Function Clone() As Object
			Return New Image(_imageInfo)
		End Function

		Public Overrides ReadOnly Property Size() As SizeF
			Get
				Return New SizeF(CType(_imageInfo.PixelWidth * 1440 / _imageInfo.HorizontalResolution, Single), CType(_imageInfo.PixelHeight * 1440 / _imageInfo.VerticalResolution, Single))
			End Get
		End Property

		Public Shared Widening Operator CType(image As Image) As XImage
			Return image._imageInfo
		End Operator
	End Class
End Class

File: Advanced/PageAndRDL/CustomPdfExport/C#/PdfRendering/Rendering/PdfContentGenerator.Image.cs
using System.Drawing;
using GrapeCity.ActiveReports.Drawing;
using PdfSharp.Drawing;

namespace GrapeCity.ActiveReports.Samples.Export.Rendering.PdfSharp
{
	partial class PdfContentGenerator
	{
		private class Image : ImageEx
		{
			private readonly XImage _imageInfo;

			public Image(XImage imageInfo)
			{
				_imageInfo = imageInfo;
			}

			public override object Clone()
			{
				return new Image(_imageInfo);
			}

			public override SizeF Size
			{
				get { return new SizeF((float)(_imageInfo.PixelWidth * 1440 / _imageInfo.HorizontalResolution), (float)(_imageInfo.PixelHeight * 1440 / _imageInfo.VerticalResolution)); }
			}

			public static implicit operator XImage(Image image)
			{
				return image._imageInfo;
			}
		}
	}
}
